core/src/main/java/org/apache/iceberg/metrics/ScanMetrics.java

```diff
@@ -127,6 +129,16 @@ public Counter positionalDeleteFiles() {
     return metricsContext().counter(POSITIONAL_DELETE_FILES);
   }
 
+  public void indexedDeleteFile(DeleteFile deleteFile) {
```

Comment:
I wonder if it would make sense to define this method in `DeleteFileIndex` rather than adding this as a public API of `ScanMetrics`?

Reply:
I wasn't sure myself. It felt to me a few similar methods like this would be appropriate to hide the complexity of populating the same values from multiple places. If we go with this one, I'd expect more methods. For instance, we could have one that accepts a matching data file and updates both the count as well as the total file size. A similar thing can be done for the array of delete files, etc.

I see a few potential benefits:
- Less metrics related code (1 line vs 14 lines in the delete index), moving the complexity to `ScanMetrics`.
- Methods can be shared (e.g. distributed planning updates exactly the same metrics as `ManifestGroup`).
- We can add more metrics implicitly without requiring changes outside of `ScanMetrics`.

That said, I don't feel strongly. How did you envision this API, @nastra?

Reply:
Let me think about this and the implications of that API and I'll reply here tomorrow

Reply:
I gave this some thought. I think we shouldn't add complexity to the `ScanMetrics` API itself in order to populate more than one metric. That just adds additional complexity to the API and all of a sudden also adds non-metrcis-related dependencies to it (like `DeleteFile` and such).

The purpose of `ScanMetrics` is to just be a data container for scan-related metrics, that are populated during scan planning, so I think it would be better to keep this API clean.

On the other hand I understand that we want to reduce metrics-related stuff in the code itself and I think we should maybe just have a **util class** that handles this? This would allow us to keep `ScanMetrics` clean but also extract metrics-related code that calls the right increments depending on a particular file type in one single place.

Reply:
A utility class sounds good, I'll create one.

Reply:
thanks @aokolnychyi